docs/how-to/create_plugin.rst

```diff
@@ -0,0 +1,131 @@
+.. _how_to_create_plugin:
+
+How to create a plugin
+======================
+
+Plugins help to bring new tools and programming languages into craft-parts.
+This document contains instructions on how to recreate the simple
+:py:class:`~craft-parts.plugins.dump_plugin.DumpPlugin` plugin.
+
+A plugin is made up of two major components: its properties and its
+functionality. The properties of a plugin are the unique pieces of
+configuration that make up the user interface of the plugin. These include
+things like specifying a Python version to use in a Python-based plugin, or
+the authentication scheme to use to connect to a remote server - it all
+depends on what your plugin does, and what it needs to do its job! The
+functionality of the plugin is the Python code that makes up its actual
+execution. The functionality of the plugin should take in configuration from
+the plugin's properties.
+
+Create your plugin file
+-----------------------
+The first step in creating a plugin is deciding where it will live. If you're
+adding a plugin directly to ``craft-parts``, then it should go in
+:file:`craft_parts/plugins`. Since we're re-creating the ``dump`` plugin,
+we'll put it in :file:`craft_parts/plugins/new_dump_plugin.py`.
+
+Imports
+~~~~~~~
+First, we'll have to pull in our imports. Since this is such a simple plugin,
+the list of necessary imports is quite small:
+
+.. code:: py
+
+    from typing import Literal
+    from overrides import override
+
+    from .base import Plugin
+    from .properties import PluginProperties
```

Comment:
I'm not sure how useful a section on imports is. Python coders know there should be imports, and scripts are open-ended. I don't see much in the way of a decision the reader needs to make here. Is there anything specific to plugins that the reader ought to do as far as imports? If not, we could probably cut this.

Reply:
The need to import `Literal` and `override` should be pretty easy to guess if we just started using it in code without mention of it, but first-time developers could get confused on `Plugin` and `PluginProperties` since they aren't from any standard library, but rather declared locally. I could move each of the imports into their relevant code blocks, though. What do you think?

Reply:
Groovy. Let's tell the reader to import `Plugin` and `PluginProperties`. You can drop a quick line about those being local and necessary for the rest of the class.

I don't think it's necessary to itemise the code blocks, though.